File: client/map.js
const Map = {
    filter: (array, func) => array.filter(func),
    map: (array, func) => array.map(func),
    forEach: (array, func) => array.forEach(func),
    shuffle: (array) => array.map(a => [Math.random(), a]).sort((a, b) => a[0] - b[0]).map(a => a[1])
}

File: client/map.lua
Map = {}

function Map.filter(array, func)
    local returnTable = {};

    for index, v in pairs(array) do
        if func(v, index) then
            table.insert(returnTable, v)
        end
    end

    return returnTable
end

function Map.map(array, func)
    local returnTable = {};

    for _, v in pairs(array) do
        table.insert(returnTable, func(v))
    end
    
    return returnTable
end

function Map.forEach(array, func)
    for index, value in pairs(array) do
        func(value, index)
    end
end

function Map.shuffle(array)
    for i = #array, 2, -1 do
        local j = math.random(i)
        array[i], array[j] = array[j], array[i]
    end

    return array
end